src/Spice86/Converters/AddressStringToSegmentedAddressConverter.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Spice86.Shared.Emulator.Memory;
+using Spice86.Shared.Utils;
+
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace Spice86.Converters;
+
+public partial class AddressStringToSegmentedAddressConverter : SegmentedAddressConverter
+{
+
+    public override object? Convert(object? value, Type targetType, object? parameter, CultureInfo culture)
+    {
+        return base.Convert(value, targetType, parameter, culture);
+    }
+
+    public override object? ConvertBack(object? value, Type targetType, object? parameter, CultureInfo culture)
+    {
+        if (value is not string str || string.IsNullOrWhiteSpace(str))
+        {
+            return new SegmentedAddress();
+        }
+
+        // Match 0x hexadecimal address
+        Match hexMatch = HexAddressRegex().Match(str);
+        if (hexMatch.Success)
+        {
+            if (uint.TryParse(hexMatch.Groups[1].Value,
+                NumberStyles.HexNumber, CultureInfo.InvariantCulture,
+                out uint address))
+            {
+                return MemoryUtils.ToSegmentedAddress(address);
+            }
+        }
+
+        // Match hexadecimal address without 0x prefix
```

Comment:
If the input can be either decimal or hexadecimal you _need_ a specifier like `0x` or `h`. 
Otherwise you have no idea how to treat "1234". It could be 1234, or it could be 4660. You can never know.

Reply:
Agree.

Reply:
Fixed.